var/spack/repos/builtin/packages/mvapich2/package.py

```diff
@@ -204,6 +204,10 @@ def setup_build_environment(self, env):
         env.unset('F90')
         env.unset('F90FLAGS')
 
+        # https://bugzilla.redhat.com/show_bug.cgi?id=1795817
+        if self.spec.satisfies('%gcc@10:'):
+            env.set('FFLAGS', '-fallow-argument-mismatch')
```

Comment:
Normally I'd prefer overriding the flag handler as you did in https://github.com/spack/spack/pull/17113, since that allows merging sets of flags together. Is there a reason you did it differently here?

I'm also wondering if there should be a constraint to only do this for `mvapich2@:2.3.4` (assuming future versions will not require this flag).

Reply:
No real reason, just because I did it this way for mpich. Should I update both to use flag handlers? Can we mix flag handlers and `setup_build_environment`?

Reply:
It's fine although the `flag_handler` approach is a bit more flexible.

> Can we mix flag handlers and setup_build_environment?

As long as you don't use the environment-flags-export option in the flag handler, there shouldn't be a conflict between using `flag_handler` and setting compiler flag environment variables directly. See Also https://spack.readthedocs.io/en/latest/packaging_guide.html?highlight=flag_handler#compiler-flags

Reply:
Sorry for the delay. I just pushed a new version using `flag_handler`.